bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java

```diff
@@ -192,17 +192,20 @@ public void onIntegratedPullRequest(PullRequest pr, Path scratchPath, Hash hash)
 
             // If prOnly is false, this is instead done when processing commits
             if (prOnly && resolve) {
-                if (issue.state() == Issue.State.OPEN) {
-                    log.info("Resolving issue " + issue.id());
-                    issue.setState(Issue.State.RESOLVED);
-                    if (issue.assignees().isEmpty()) {
-                        var username = findIssueUsername(commit, scratchPath);
-                        if (username.isPresent()) {
-                            var assignee = issueProject.issueTracker().user(username.get());
-                            assignee.ifPresent(hostUser -> issue.setAssignees(List.of(hostUser)));
-                        }
+                log.info("Resolving issue " + issue.id() + " from state " + issue.state());
```

Comment:
We should keep the log message and the issue.setState call together. I think moving lines 203-207 up here would be best to keep the flow as similar to before as we can. Same applies in the other method.

Reply:
Sure